### ISSUE TITLE
(Reverse) Trail items lose their individual offsets on reverse animation.

### ISSUE THREAD

**Issue by Schmerb** [react-spring/react-spring#667]
## 🐛 Bug Report

When using the `Trail` render-prop to animate a list of items, everything works as expected until reversing the animation via `reverse={toggle}` prop. This does appear to reverse the order of animation; however, the offset between each item is so minimal that it almost appears to close all items synchronously. 

Default order (`reverse={false}`) -->  items animate in with offsets as expected

Reverse order (`reverse={true}`) --> items appear to animate out all at the same time with very minimal offsets, if any. 

## To Reproduce

Create a simple `Trail` component and toggle the reverse prop on animation close. 

```   
const items = [...]
 const DX = 40;
<Trail
          native
          reverse={!toggle}
          config={config.stiff}
          initial={null}
          items={items}
          keys={item => item.key}
          from={{ opacity: 0, x: -DX }}
          to={{ opacity: toggle ? 1 : 0.25, x: toggle ? 0 : DX }}
        >
          {item => ({ x, opacity }) => (
            <animated.div
              style={{
                opacity,
                transform: x.interpolate(x => `translate3d(${x}px,0,0)`),
              }}
            >
              {item}
            </animated.div>
          )}
        </Trail>
```

## Expected behavior

The `Trail` items should animate out in *reverse* order but with the same offsets between each item as when they enter. 

## Link to repro 

[CodeSandbox Bug Demo](https://codesandbox.io/s/oqrolk2055) 

## Environment

- `react-spring` v8.0.20
- `react` v16.8.x (or `react-native` v0.58.x)

**Comment by aleclarson** [comment]
Hey, thanks for the bug report. Can you try with `react-spring@next`?

**Comment by Schmerb** [comment]
Sure thing. I just tried using `9.0.0-beta.8` and still same issue on reverse, no offsets

**Comment by aleclarson** [comment]
This is fixed in v9.0.0-beta.9

https://codesandbox.io/s/react-spring-issue-667-dkoe3

**Comment by Schmerb** [comment]
Excellent! Thank you for your time and effort
